sklearn/metrics/_plot/roc_curve.py

```diff
@@ -0,0 +1,151 @@
+from .. import auc
+from .. import roc_curve
+
+from ...utils import check_matplotlib_support
+
+
+class RocCurveVisualizer:
+    """ROC Curve visualization.
+
+    It is recommend to use `sklearn.metrics.plot_roc_curve` to create this
+    object.
+
+    Read more in the :ref:`User Guide <visualizations>`.
+
+    Parameters
+    ----------
+    fpr : ndarray
+        False positive rate.
+    tpr : ndarray
+        True positive rate.
+    auc_roc : float
+        Area under ROC curve.
+    estimator_name : str
+        Name of estimator.
+
+    Attributes
+    ----------
+    line_ : matplotlib Artist
+        ROC Curve.
+    ax_ : matplotlib Axes
+        Axes with ROC Curve
+    figure_ : matplotlib Figure
+        Figure containing the curve
+    """
+
+    def __init__(self, fpr, tpr, auc_roc, estimator_name):
+        self.fpr = fpr
```

Comment:
Please document these attributes or make them private. Or add a note indicating that all parameters are also attributes if you want to avoid documenting twice.

Reply:
In our estimators, we store the `__init__` parameters the same way and do not document the parameters as attributes.

Reply:
But these aren't estimators. This behaviour is documented for estimators. Not for this class. A short note doesn't hurt.

Reply:
Update docstring with:

> It is recommend to use `sklearn.metrics.plot_roc_curve` to create a visualizer. All parameters are stored as attributes.